sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsAtomicPartitionManagement.java

```diff
@@ -82,4 +86,19 @@ void createPartitions(
    * @return true if partitions were deleted, false if any partition not exists
    */
   boolean dropPartitions(InternalRow[] idents);
+
+  /**
+   * Drop an array of partitions atomically from table, and completely remove partitions data
+   * by skipping a trash even if it is supported.
+   * <p>
+   * If any partition doesn't exists,
+   * the operation of purgePartitions need to be safely rolled back.
+   *
+   * @param idents an array of partition identifiers
+   * @return true if partitions were deleted, false if any partition not exists
+   * @throws NoSuchPartitionException If any partition identifier to alter doesn't exist
+   * @throws UnsupportedOperationException If partition purging is not supported
+   */
```

Comment:
let's add `@since 3.2.0`

Reply:
Actually, I assumed to port this to `branch-3.1` since the interface has not been released yet, see your comment https://github.com/apache/spark/pull/30776#discussion_r543130529

Reply:
Previously, we need to add a parameter, and it's better to change the un-released API directly.  Now, we are adding a new method,  then I don't think it's qualified to backport to 3.1.